packages/hardhat-core/src/internal/hardhat-network/provider/vm/dual.ts

```diff
@@ -59,6 +69,24 @@ export class DualModeAdapter implements VMAdapter {
       selectHardfork
     );
 
+    // if the fork node config doesn't specify a fork block number, then the
+    // ethereum-js VM used the latest block number from the forked chain. here
+    // we get that number from that VM so it can be passed into the rethnet VM;
+    // otherwise, the rethnet VM's determination of "latest" might be a later
+    // number, resulting in a different fork block.
```

Comment:
I found this explanation a bit hard to understand. This is my attempt to re-write it, but feel free to ignore it.

```suggestion
    // If the fork node config doesn't specify a block number, then the
    // ethereumjs adapter will fetch and use the latest block number. We re-use
    // that value here; otherwise, rethnet would also fetch it and we could have
    // a race condition if the latest block changed in the meantime.
```

Reply:
Replace Gene's explanation with yours. I believe it's correct, but @feuGeneA could you please verify that Franco's comments still carry the same meaning as you intended.

Reply:
LGTM!